astronomer/providers/amazon/aws/sensors/s3.py

```diff
@@ -89,27 +91,37 @@ def execute(self, context: Context) -> None:
             else:
                 raise e
         if not poke:
-            self.defer(
-                timeout=timedelta(seconds=self.timeout),
-                trigger=S3KeyTrigger(
-                    bucket_name=cast(str, self.bucket_name),
-                    bucket_key=self.bucket_key,
-                    wildcard_match=self.wildcard_match,
-                    check_fn=self.check_fn,
-                    aws_conn_id=self.aws_conn_id,
-                    verify=self.verify,
-                    poke_interval=self.poke_interval,
-                    soft_fail=self.soft_fail,
-                ),
-                method_name="execute_complete",
-            )
+            self._defer()
+
+    def _defer(self):
+        self.defer(
+            timeout=timedelta(seconds=self.timeout),
+            trigger=S3KeyTrigger(
+                bucket_name=cast(str, self.bucket_name),
+                bucket_key=self.bucket_key,
+                wildcard_match=self.wildcard_match,
+                check_fn=self.check_fn,
+                aws_conn_id=self.aws_conn_id,
+                verify=self.verify,
+                poke_interval=self.poke_interval,
+                soft_fail=self.soft_fail,
+                should_check=self.should_check,
+            ),
+            method_name="execute_complete",
+        )
 
     def execute_complete(self, context: Context, event: Any = None) -> bool | None:
         """
         Callback for when the trigger fires - returns immediately.
         Relies on trigger to throw an exception, otherwise it assumes execution was
         successful.
         """
+        if event["status"] == "running":
+            found_keys = self.check_fn(event["files"])
+            if found_keys:
+                return None
+            else:
+                self._defer()
```

Comment:
This is just a function callable here. So why is it required to defer here? I didn't understand this @pankajastro